api/api_errors.go

```diff
@@ -160,3 +165,42 @@ func NewErrExecutionReverted(reason string) *ErrExecutionReverted {
 		Data:    reason,
 	}
 }
+
+type ErrNullRound struct {
+	Epoch   int64
+	Message string
+}
+
+func NewErrNullRound(epoch int64) *ErrNullRound {
+	return &ErrNullRound{
+		Epoch:   epoch,
+		Message: fmt.Sprintf("requested epoch was a null round (%d)", epoch),
+	}
+}
+
+func (e *ErrNullRound) Error() string {
+	return e.Message
+}
+
+func (e *ErrNullRound) FromJSONRPCError(jerr jsonrpc.JSONRPCError) error {
+	if jerr.Code != ENullRound {
+		return fmt.Errorf("unexpected error code: %d", jerr.Code)
+	}
+
+	epoch, ok := jerr.Data.(float64)
+	if !ok {
+		return fmt.Errorf("expected number data in null round error, got %T", jerr.Data)
+	}
+
+	e.Epoch = int64(epoch)
+	e.Message = jerr.Message
+	return nil
+}
+
+func (e *ErrNullRound) ToJSONRPCError() (jsonrpc.JSONRPCError, error) {
+	return jsonrpc.JSONRPCError{
+		Code:    ENullRound,
+		Message: e.Message,
+		Data:    e.Epoch,
+	}, nil
+}
```

Comment:
```suggestion
}

// Is performs a non-strict type check, we only care if the target is an ErrNullRound
// and will ignore the contents (specifically there is no matching on Epoch).
func (e *ErrNullRound) Is(target error) bool {
       _, ok := target.(*ErrNullRound)
       return ok
}
```

We're going to need our own `Is` implementation because `errors.Is` does a direct match, which is fine for sentinel errors but this isn't a sentinel. In addition, we're going to do it non-strict. A standard form of this might check that `Epoch` was the same, but all we care about is this as as if it's a sentinel error.